R/vctr.R

```diff
@@ -157,8 +157,7 @@ format.vctrs_vctr <- function(x, ...) {
 
 #' @export
 `[.vctrs_vctr` <- function(x, i, ...) {
-  check_dots_empty_s3_consistency(...)
-  vec_slice_native(x, maybe_missing(i, TRUE))
+  vec_index(x, i, ...)
```

Comment:
Interestingly this makes it clear that `[` is not type-stable. I think that's correct, so nothing needs to change, but it's worth noting.

Reply:
I tend to think of it as dimensionality-stable, but not dimension-stable (which makes it not type-stable).